consumer/consumererror/signalerrors.go

```diff
@@ -24,20 +33,44 @@ func (err retryable[V]) Data() V {
 	return err.data
 }
 
+// Delay returns the time duration before the telemetry data should
+// be resent to the destination.
+func (err retryable[V]) Delay() time.Duration {
+	return err.delay
+}
+
+// RetryOption allows adding data to a retryable error,
+// e.g. the duration before sending should be retried.
+type RetryOption func(err *retryableCommon)
```

Comment:
Why not `RetryableOption` given that it potentially can be any option applied to retryable errors?

Reply:
Makes sense to me, done.